src/UI/Component/Icon/Factory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+namespace ILIAS\UI\Component\Icon;
+/**
+ * This is how a factory for icons looks like.
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *
+	 *   composition: >
+	 *     A Standard Icon is displayed as a block-element with a background-graphic.
+	 *     By default, a fallback icon will be rendered; this is until a
+	 *     background image is defined in the icon's CSS-class.
+	 *   rivals:
+	 *     1: Custom Icons are constructed with a path to an (uploaded) image.
```

Comment:
Icons MUST use SVG as an graphic.

Icons MUST have a transparent background so they could be put on all kinds of backgrounds.